text/0033-JAM-Service-for-Validating-Ethereum-Optimistic-Rollups.md

```diff
@@ -0,0 +1,200 @@
+
+This approach, however, incurs a cost: a 7-day exit window to accommodate fraud proofs. JAM Service can eliminate this long exit window by validating optimistic rollups.
+
+## Motivation
+
+JAM is intended to host rollups rather than serve end users directly.
+
+A secure JAM service for Optimistic Rollups will expand JAM's service scope to include optimistic rollups and enhance their appeal with JAM's finality.
+
+This expanded focus on rollups will increase CoreTime demand, making JAM attractive to both existing and new optimistic rollups by reducing exit windows.
+
+This JAM Service converts optimistic rollups into cynical rollups, delivering a user experience advantage that is marketable. With popular CEXes (Coinbase, Kraken) adopting OP Stack, this improvement is highly visible to retail users, turning ORUs into CRUs "Secured by JAM" for a competitive edge.
+
+## Requirements
+
+1. Securing optimistic rollups with a JAM Service should be practical and require minimal changes by the ORU operator.
+2. Securing Polkadot rollups with a JAM Service should **not** be affected.
+
+## Stakeholders
+
+1. Optimistic Rollup Operators seeking shorter exit windows, lower costs, higher scalability, and DA.
+2. Web3 developers wanting to create applications on cynical rollups secured by JAM.
+3. DOT Holders aiming to increase CoreTime demand.
+
+# JAM Service Design
+
+## Overview
+
+Rollups produce a sequence of blocks ${\bf B}$ with headers ${\bf H}$. The header ${\bf H}$ contains a parent header hash ${\bf H}_p$ and a state root ${\bf H}_{r}$, which represents the global state after applying all block transactions. The transactions trie root is unnecessary here; validating the state trie root alone is sufficient for securing rollups.  
+
+This JAM Service strategy, as hinted in the JAM Gray Paper, aggregates storage proofs of altered smart account/contract states into a chain of work packages. These validate an uninterrupted chain of an optimistic rollup's blocks against the state root ${\bf H}_r$.
+
+Instead of relying on a *promise* that the state root is correct unless proven otherwise, JAM validates it using storage proofs for each contract update across $N$ blocks.
+
+The rollup operator submits headers and Merkle proofs in a chain of work packages $...,  p_{n-1}, p_{n}, p_{n+1}, ...$ corresponding to the rollup chain, grouping $N$ blocks of data in each package. Typically, each work package $p_n$ requires a prerequisite work package $p_{n-1}$.
+
+## Refine:
+
+#### Key Input: Work Packages
+
+In JAM's refine code, work package content is as follows:
+
+| JAM Refine               | Content |
+| --- | --- |
+| *Work Package* $p_n \in \mathbb{P}$     | Data submitted by ORU operator for validation of blocks $i$ through $j$ |
```

Comment:
it's important to support partial validation in a WP, i.e. to spread the validation of a rollup batch across multiple WPs. The trend of rollups is to have large sequencers and produce heavy blocks. With the current formulation, you'd only support rollups whose batches fit into a single WP.

Note that WPs will require state tree proofs and will likely be bounded more by data than by compute.

Reply:
Will give it a shot, thank you!